app/src/ui/multi-commit-operation/choose-branch/merge-choose-branch-dialog.tsx

```diff
@@ -127,12 +125,17 @@ export class MergeChooseBranchDialog extends BaseChooseBranchDialog {
 
     if (this.state.selectedBranch !== branch) {
       this.commitCount = 0
+      return
     }
 
-    this.updateMergeStatusPreview(branch)
+    this.updateMergeStatusPreview(branch, mergeStatus)
   }
 
-  private updateMergeStatusPreview(branch: Branch) {
+  private updateMergeStatusPreview(
+    branch: Branch,
+    mergeStatus: MergeTreeResult
+  ) {
+    this.mergeStatus = mergeStatus
```

Comment:
Don't need to update the class variable unless updating the state... probably should refactor to store these in the state and just update the preview on state update... but that PR may be pretty hefty.. so just focusing on issue at hand with this pr.